test/decorator-annotator_test.ts

```diff
@@ -28,68 +30,82 @@ function sources(sourceText: string): Map<string, string> {
-        const program = testSupport.createProgram(sources(sourceText));
-        const sourceMapper = new DefaultSourceMapper(testCaseFileName);
-        const {output, diagnostics} = convertDecorators(
-            program.getTypeChecker(), program.getSourceFile(testCaseFileName), sourceMapper);
-        if (!allowErrors) expect(diagnostics).to.be.empty;
-        verifyCompiles(output);
-        return {output, diagnostics, sourceMap: sourceMapper.sourceMap};
-      }
-
-      function expectUnchanged(sourceText: string) {
-        expect(translate(sourceText).output).to.equal(sourceText);
-      }
-
-      it('generates a source map', () => {
-        const {output, sourceMap} = translate(`
-/** @Annotation */ let Test1: Function;
-@Test1
-export class Foo {
-}
-let X = 'a string';`);
-        const rawMap = sourceMap.toJSON();
-        const consumer = new SourceMapConsumer(rawMap);
-        const lines = output.split('\n');
-        const stringXLine = lines.findIndex(l => l.indexOf('a string') !== -1) + 1;
-        expect(consumer.originalPositionFor({line: stringXLine, column: 10}).line)
-            .to.equal(6, 'string X definition');
-      });
-
-      describe('class decorator rewriter', () => {
-        it('leaves plain classes alone', () => {
-          expectUnchanged(`class Foo {}`);
-        });
-
-        it('leaves un-marked decorators alone', () => {
-          expectUnchanged(`
-          let Decor: Function;
+    const transformerHost: tsickle.TsickleHost = {
+      shouldSkipTsickleProcessing: (filePath) => !sources(sourceText).has(filePath),
+      pathToModuleName: cliSupport.pathToModuleName,
+      shouldIgnoreWarningsForPath: (filePath) => false,
+      fileNameToModuleId: (filePath) => filePath,
+      transformDecorators: true,
+      googmodule: true,
+      es5Mode: false,
+      untyped: false,
+    };
+
+    const files = new Map<string, string>();
+    const {diagnostics} = tsickle.emitWithTsickle(
+        program, transformerHost, host, compilerOptions, undefined, (path, contents) => {},
+        undefined, undefined, {beforeTs: [generateAstPrintingTransform(files)]});
+
+    if (!allowErrors) {
+      // tslint:disable-next-line:no-unused-expression
+      expect(diagnostics, tsickle.formatDiagnostics(diagnostics)).to.be.empty;
+    }
+
+    return {output: files.get(testCaseFileName)!, diagnostics};
+  }
+
+  function prettyPrint(sourceText: string) {
+    return ts.createPrinter().printFile(
+        ts.createSourceFile('', sourceText, compilerOptions.target!));
+  }
+
+  function normalizeQuotes(sourceText: string) {
+    return sourceText.replace(/"/g, `'`);
+  }
+
+  function expectUnchanged(sourceText: string) {
+    expectEqual(sourceText, sourceText);
+  }
+
+  function expectEqual(sourceText: string, expected: string) {
+    expect(normalizeQuotes(translate(sourceText).output))
+        .to.equal(normalizeQuotes(prettyPrint(expected)));
```

Comment:
can't you make the expected you're passing in here match the pretty print? That'd safe some code and avoid a potential pitfall/source of error.

Reply:
I'd be happy to do that in a follow up PR, but for this one, I wanted the golden diff to be as small as possible so we could actually be confident that the tests were still asserting the right stuff.